toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -247,8 +250,9 @@ func CustomizeImage(ctx context.Context, buildDir string, baseConfigPath string,
 	)
 	defer func() {
 		if err != nil {
-			span.RecordError(fmt.Errorf("image customization failed"))
-			span.SetStatus(codes.Error, "image customization failed")
+			category := GetErrorCategory(err)
```

Comment:
Lets record the category and the error code, separately but part of the RecordError() function if possible.